packages/overlay/src/OverlayBase.ts

```diff
@@ -0,0 +1,908 @@
+        ) {
+            return;
+        }
+
+        const triggerRoot = trigger.getRootNode();
+        const contentRoot = this.elements[0].getRootNode();
+        const overlayRoot = this.getRootNode();
+        if (triggerRoot == overlayRoot) {
+            const releaseAriaDescribedby = conditionAttributeWithId(
+                trigger,
+                'aria-describedby',
+                [this.id]
+            );
+            this.releaseAriaDescribedby = () => {
+                releaseAriaDescribedby();
+                this.releaseAriaDescribedby = noop;
+            };
+        } else if (triggerRoot === contentRoot) {
+            this.elementIds = this.elements.map((el) => el.id);
+            const appliedIds = this.elements.map((el) => {
+                if (!el.id) {
+                    el.id = `${this.tagName.toLowerCase()}-helper-${crypto
+                        .randomUUID()
+                        .slice(0, 8)}`;
+                }
+                return el.id;
+            });
+            const releaseAriaDescribedby = conditionAttributeWithId(
+                trigger,
+                'aria-describedby',
+                appliedIds
+            );
+            this.releaseAriaDescribedby = () => {
+                releaseAriaDescribedby();
+                this.elements.map((el, index) => {
+                    el.id = this.elementIds[index];
+                });
+                this.releaseAriaDescribedby = noop;
+            };
+        }
+    }
+
+    private handlePointerdown = (event: PointerEvent): void => {
+        if (!this.triggerElement) return;
+        if (event.button !== 0) return;
+        const triggerElement = this.triggerElement as HTMLElement;
+        this.longpressed = false;
+        triggerElement.addEventListener('pointerup', this.handlePointerup);
+        triggerElement.addEventListener('pointercancel', this.handlePointerup);
+        this.longressTimeout = setTimeout(() => {
+            if (!triggerElement) return;
+            triggerElement.dispatchEvent(
+                new CustomEvent<LongpressEvent>('longpress', {
+                    bubbles: true,
+                    composed: true,
+                    detail: {
+                        source: 'pointer',
+                    },
+                })
+            );
+        }, LONGPRESS_DURATION);
+    };
+
+    private handlePointerup = (): void => {
+        clearTimeout(this.longressTimeout);
+        if (!this.triggerElement) return;
+        if (this.longpressed) {
+            this.open = true;
+        }
+        setTimeout(() => {
+            this.longpressed = false;
+        });
+        const triggerElement = this.triggerElement as HTMLElement;
+        triggerElement.removeEventListener('pointerup', this.handlePointerup);
+        triggerElement.removeEventListener(
+            'pointercancel',
+            this.handlePointerup
+        );
+    };
+
+    /**
+     * @private
+     */
```

Comment:
NIT: Why doc marking it as @private here when it's protected?

Reply:
Generally, we've done this to prevent these methods from surfacing in the API table, I'll double check that's the case here and add a comment.